### DISCUSSION TITLE
Chinese document translation

### DISCUSSION THREAD

**Discussion by ohmyide** [eclipse-theia/theia#10957] [Ideas]
Theia is very popular in China, but there is a lack of Chinese documentation, would you mind me translating it? The content includes the official website and wiki. In fact, I have already translated part of it. I wonder if you allow it

**Comment by msujew** [comment]
cc @JonasHelming @tsmaeder @paul-marechal Your thoughts on that? Other popular open source frameworks also translate their websites into other languages, such as [kubernetes](https://kubernetes.io/zh/).

**Reply by vince-fugnitto** [reply]
@msujew how would we plan on maintaining multiple versions of our documentation, and ensuring that the documentation is accurate and up-to-date for different languages (will it be something automated)? Perhaps a dumb question but wouldn't the browser's translation utilities be sufficient in translating the existing documentation for most languages?

**Reply by msujew** [reply]
Absolutely no clue, I just posted to make the dev-leads aware of this. By the way, I'm not totally sure how browsers in China work, but I believe only Chrome does automatic website translation (using Google Translate) and IIRC that's not available in China.

**Comment by JonasHelming** [comment]
@ohmyide : Thank you for your suggestion! Would you also be willing to continuously contribute to the maintenance of the translated doc?
and, could you comment on the questions raised above, please?

**Comment by tsmaeder** [comment]
Like Jonas, I'm not against having translations (why stop at Mandarin, if we can have Kalmyk ;-)). However, the current group of committers (https://projects.eclipse.org/projects/ecd.theia/who) does not have the knowledge to do that. The Mandarin-Speaking community would have to step up to that plate.
The problem is not so much doing an initial translation, but keeping it current. That would need some commitment from contributors.

Instead of doing a translated version of the website, how about maintaining a collection of Theia-related content in Mandarin and other languages? That would lower the burden of expectations for completeness and currency.

**Comment by ohmyide** [comment]
In China, the chrome browser accounts for about 60%, the translation software is not very convenient, and the translation quality is not very good.

In fact, in China we really need a collection of Chinese documents about Theia-related, and I really want to spread and maintain this collection.

I know your concerns and agree with them. If I promise to do this, it will be guaranteed to be done, which is also my personal hobby.

**Reply by JonasHelming** [reply]
Thank you, I have put this on the agenda for our [next dev call](https://github.com/eclipse-theia/theia/wiki/Dev-Meetings) and I will post the results here!

**Reply by ohmyide** [reply]
👌

**Reply by JonasHelming** [reply]
We discussed this in the dev call today. We would like to support people in any particular language requested to enable more people adopting and contributing to Theia. So it would be great if you could contribute to that!
To address concerns, we agreed on the following:

1. The general project language is English, so communication on tickets and especially the reference documentation will be English
2. Support for other languages can only be offered if contributors continuously work on it and maintain it.
3. Committers cannot ensure the quality on other languages, especially if no committer speaks them (which is currently the case for Chinese)
4. Therefore, we need to add a disclaimer to other languages: "The official reference documentation of Eclipse Theia is in English. This translation is fully based on contributions. Due to lack of language skills, the Theia core team cannot ensure the quality or completeness of this translation. If you find any issues, please report them. Fixes will fully depend on contributions, though.
5. The translations should live in a separate area, i.e. you can access them from the website or the reference documentation using a link that always shows the disclaimer.

If the points above make sense to you, we would love to get an example contribution, maybe starting with the documentation on the website. It should show how the translation could look like and how it would be embedded into the website (following point 5).
Looking forward to your contribution!

**Reply by ohmyide** [reply]
I agree with the above principles.

[theia-website](https://github.com/ohmyide/theia-website)  was created by gatsby.

I will think about how to present the  Chinese translation, and then further communicate and confirm with you.

**Reply by JonasHelming** [reply]
Sounds great, looking forward to your contribution!

**Reply by ohmyide** [reply]
@JonasHelming  Sorry I'm late.
In fact, I finished the translation three months ago: https://github.com/ohmyide/theia-website-zh
Here is the temporary website：http://theia.ohmyide.com/
And I added the following disclaimer in the footer：
<img width="763" alt="image" src="https://user-images.githubusercontent.com/96159101/187137523-5446a24e-25cf-441c-b262-3ce14a233865.png">

As above I would like to hear your suggestions.

**Reply by ohmyide** [reply]
@msujew  @tsmaeder  @vince-fugnitto 
Looking forward to your response.  😁

**Reply by planger** [reply]
@ohmyide Awesome! :100: I just wanted to let you know that @JonasHelming is on vacation this week. He'll get back to you next week.

**Reply by JonasHelming** [reply]
@ohmyide Sorry for the late reply, I am just back from vacation. Your translation looks great although I obviously do not understand a single word :-)
Whats missing from my POV would be a mechanism for the language switcher. Ideally, this would be seemless for the English content i.e. we ideally do not have to touch the existing website to much. Can you propose some solution here?
Furthermore, I would like to move the disclaimer rather to the top of the site to make it more visible rather then having it in the footer.

Thank you for your great work!

**Comment by ohmyide** [comment]
Many open source projects have Chinese documentation, which helps many people, such as:

inversify：https://doc.inversify.cloud/zh_cn/
kubernetes：https://kubernetes.io/zh/
webpack：https://webpack.docschina.org/concepts/
vuejs：https://staging-cn.vuejs.org/
lodash：https://www.lodashjs.com/
